lib/jekyll/tags/include.rb

```diff
@@ -21,6 +21,10 @@ class IncludeTag < Liquid::Tag
         (?<params>.*)
       !x
 
+      FULL_VALID_SYNTAX = %r!\A\s*(?:#{VALID_SYNTAX}(?=\s|\z)\s*)*\z!
```

Comment:
* `(?=\s|\z)\s*)*\z` It seems like there might be multiple ways for this to match. That's bad for performance.
 * What is the purpose of the `(?=\s|\z)`?
 * This matches the empty string (but `VALID_SYNTAX` does not). Is that intentional?

I see you are just hoisting this from `validate_params` so maybe these issues should be addressed separately?

Reply:
- `VALID_SYNTAX` is probably a poorly named constant because its actually intended to just validate `params` in the include tag instead of the entire tag markup..
- `FULL_VALID_SYNTAX` is a superset of `VALID_SYNTAX` so its allowed that the former matches more than the latter..
- `(?=\s|\z)` is to ensure that `params` is either singular `foo='bar'` or space-separated
 `foo='bar' alpha="beta"`. `foo='bar'alpha="beta"` wouldn't match.

Reply:
> `(?=\s|\z)` is to ensure that params is either singular foo='bar' or space-separated

So we can get rid of the following `\s*` and `\z`?

`%r!\A\s*(?:#{VALID_SYNTAX}(?=\s|\z)+)+!`?

"Each repetition of `VALID_SYNTAX` must be followed by either spaces or the end of the string."